util-hadoop/src/main/java/com/google/cloud/hadoop/util/CredentialFromAccessTokenProviderClassFactory.java

```diff
@@ -115,6 +117,27 @@ public static Credential credential(
     return null;
   }
 
+  /**
+   * Generate a {@link Credential} from the internal access token provider based on the service
+   * account to impersonate.
+   */
+  public static Optional<Credential> credential(
```

Comment:
Seems like this method does not belong to this class. Can we move this factory method inside `GoogleHadoopFileSystemBase` class if there are no generic credential factory where we can put it?